pkg/util/strategicpatch/diff.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategicpatch
+
+import (
+	"io"
+	"strings"
+)
+
+type diffWriter struct {
+	Writer io.Writer
```

Comment:
The short answer is that if you eliminate `Writer` and just do `io.Writer`, the struct becomes "embedded" and you get access to the io.Writer methods directly on your struct without having to write methods to forward the calls. For a longer, more detailed answer you can read https://golang.org/doc/effective_go.html#embedding